x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -869,3 +904,36 @@ function pipeMigrations(...migrations: AlertMigration[]): AlertMigration {
   return (doc: SavedObjectUnsanitizedDoc<RawRule>) =>
     migrations.reduce((migratedDoc, nextMigration) => nextMigration(migratedDoc), doc);
 }
+
+/**
+ * This creates a migration map that applies search source migrations to legacy es query rules
+ */
+function getEsQueryAlertSearchSourceMigrations(
+  searchSourceMigrations: MigrateFunctionsObject
+): MigrateFunctionsObject {
+  return mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (
+        migrateSerializedSearchSourceFields: MigrateFunction<SerializedSearchSourceFields>
+      ): MigrateFunction =>
+      (state) => {
```

Comment:
We should wrap all of these calls with `executeMigrationWithErrorHandling` and `createEsoMigration` similar to `executeMigrationWithErrorHandling(migrationRules820, '8.2.0')`.

The `executeMigrationWithErrorHandling` function is important so we properly handle decryption errors that are thrown.

The `createEsoMigration` function is important so we don't corrupt the encrypted saved object, the function will decrypt / apply your modifications / re-encrypt.

Reply:
Concur with Mike, especially `executeMigrationWithErrorHandling()`, as that's the sole place we log errors to `SavedObjectMigrationContext`.  We need errors from search source migrations logged as well, for problem diagnosis, otherwise I suspect those errors are going to be hard to identify.   Doesn't necessarily need to go through `executeMigrationWithErrorHandling()` if the signatures don't work, or whatever, but should log to that migration context anyway.  I _assume_ that will show up somewhere we can see it in Kibana logs ...

Reply:
Wrapped.

Reply:
Thanks! 🙏